Modify AOMP scripts to have the capability of building ASan for AMDGP…

ampandey-AMD opened the pull request:
…U Targets.

estewart08 committed:
Add manifest for 15.0-3 release.

ampandey-AMD committed:
Modify AOMP scripts to have the capability of building ASan for AMDGPU Targets.

ampandey-AMD committed:
Modify AOMP scripts to have the capability of building ASan for AMDGPU Targets.

ampandey-AMD committed:
Modify AOMP scripts to have the capability of building ASan for AMDGPU Targets.

ampandey-AMD committed:
Modify AOMP scripts to have the capability of building ASan for AMDGPU Targets.

estewart08 committed:
Bump AOMP version to 16.0-0.

dpalermo committed:
Babelstream: add CCC_OVERRIDE_OPTIONS check before setting -DOMP_TARGET_FAST_DOT

Check if new runtime is disabled via:
    CCC_OVERRIDE_OPTIONS='+-fno-openmp-target-new-runtime'

Don't set -DOMP_TARGET_FAST_DOT if new runtime is disabled.

estewart08 committed:
aomp_common_vars - add check for AOMP_MAJOR_VERSION 16.

estewart08 commented on bin/aomp_common_vars:
```diff
@@ -373,6 +373,12 @@ else
 fi
 
+# Sanitizer Build environment flags
+if [ ! -z "$1" ] && [ "$1" == "asan" ]; then
```
Instead of "! -z" can you use "-n" just for readability purposes? Am I missing where "asan" is passed in?

ampandey-AMD replied:
"asan" parameter is passed to enable the special ASAN environment variables to trigger the AOMP ASan build. The "asan" is parameter is passed to distinguish between the normal AOMP build and AOMP ASan build.

ampandey-AMD replied:
Closing this pull request as I had created the new one.

estewart08 committed:
Add AOMP 16.0 manifests.

estewart08 committed:
Add test manifest for AOMP 16.

dhruvachak committed:
Added test with split target construct with an inner private clause

carlobertolli committed:
Extend dynamic_allocators example and fix previous code.

estewart08 committed:
[examples] - Update asan examples to work with AOMP and ROCm.

estewart08 committed:
[examples] - Temporarily add -O2 to fortran examples.

simple_offload and declare_target fail at the default
opt level (-O0) on non-gfx90a GPUs.

estewart08 committed:
Update docs for 15.0-3 release.

estewart08 committed:
Fixing more version checks in aomp_common_vars.

These version checks will be cleaned up in the
future as we do not need to support these older
AOMP builds anymore.

estewart08 committed:
Correct build_fixups.sh to only clean part of libexec.

rocprofiler needs the python scripts in libexec/rocprofiler.

estewart08 committed:
Update changelog for 15.0-3.

nicebert committed:
Adds driver_test to openmp examples for -emit-llvm -S

estewart08 committed:
Update SOURCEINSTALL.md for 15.0-3 release.

dhruvachak committed:
Added test for no-loop codegen with immediate and nested continue.

dhruvachak committed:
Remove extra semicolon in atomic_vs_reduction test

gregrodgers committed:
change to O3 to pickup xteam reduction. Also remove num_teams and thread_limit to pick up xteam reduction for performance

gregrodgers committed:
use -O3 to get fast xteam reduction

ampandey-AMD committed:
Small Fixes in Makefile
